File: THCS/2023_2024/LaoCai_BaoThang_2324/Cau1/CAU01.py
import math

inp_file_name = "CAU01"

def main():
    if inp_file_name != "":
        s = inp_file_name + ".INP"
        freopen_in = open(s, "r")
        s = inp_file_name + ".OUT"
        freopen_out = open(s, "w")

    # Đọc input từ file
    n = int(freopen_in.readline().strip())
    a = [0] * (n + 1) 

    # Tính toán t giá trị
    for i in range(1, n + 1):
        t = 1
        c = int(math.sqrt(i))
        for j in range(2, c + 1):
            if i % j == 0:
                t += j
                if j != i // j:
                    t += i // j
        if i > 1:
            t += i
        a[i] = t

    # Đếm số cặp
    d = 0
    for i in range(1, n + 1):
        for j in range(i + 1, n + 1):
            if (a[i] == j) or (a[j] == i):
                d += 1

    # In kết quả ra file .out
    print(d, file=freopen_out)

    # Đóng file sau khi hoàn thành
    freopen_in.close()
    freopen_out.close()

if __name__ == "__main__":
    main()

File: THCS/2023_2024/LaoCai_BaoThang_2324/Cau1/CAU01.cpp
#include <bits/stdc++.h>
using namespace std;

char* inp_file_name = "CAU01";

int main() {
    if (strcmp(inp_file_name, "") != 0) {
        char s[100];
        strcpy(s, inp_file_name);
        strcat(s, ".INP");
        freopen(s, "r", stdin);  // Đọc từ file input
        strcpy(s, inp_file_name);
        strcat(s, ".OUT");
        freopen(s, "w", stdout); // Ghi ra file output
    }

    // Đọc input từ file
    int n;
    cin >> n;
    vector<int> a(n + 1, 0); 

    // Tính toán t giá trị
    for (int i = 1; i <= n; i++) {
        int t = 1;
        int c = sqrt(i);
        for (int j = 2; j <= c; j++) {
            if (i % j == 0) {
                t += j;
                if (j != i / j) {
                    t += i / j;
                }
            }
        }
        if (i > 1) {
            t += i;
        }
        a[i] = t;
    }

    // Đếm số cặp
    int d = 0;
    for (int i = 1; i <= n; i++) {
        for (int j = i + 1; j <= n; j++) {
            if (a[i] == j || a[j] == i) {
                d++;
            }
        }
    }

    // In kết quả ra file .out
    cout << d << endl;

    return 0;
}
